src/sage/combinat/integer_vectors_mod_permgroup.py

```diff
@@ -775,6 +785,79 @@ def __iter__(self):
             else:
                 return SF.elements_of_depth_iterator(self._sum)
 
+    def cardinality(self):
+        r"""Return the number of integer vectors in the set.
+
+        The number is computed using the cycle index theorem, which is
+        faster than listing the vectors.
+
+        EXAMPLES::
+
+            sage: G = PermutationGroup([], domain=[1,2,3])
+            sage: IntegerVectorsModPermutationGroup(G, 5).cardinality()
+            21
+
+            sage: G = PermutationGroup([(1,2)])
+            sage: IntegerVectorsModPermutationGroup(G, 1000).cardinality()
+            501
+
+            sage: G = PermutationGroup([(1,2,3)])
+            sage: I = IntegerVectorsModPermutationGroup(G, sum=10, max_part=5)
+            sage: I.cardinality()
+            7
+
+            sage: G = SymmetricGroup(10)
+            sage: I = IntegerVectorsModPermutationGroup(G, max_part=1)
+            sage: I.cardinality()
+            11
+        """
+        G = self._permgroup
+        k = G.degree()          # Vector length
+        d = self._sum           # Required total
+        m = self._max_part      # Max of one entry, -1 for no limit
+        if m == -1:
+            m = d               # Any entry cannot exceed total
+        if k == 0:
+            # Special case: Empty vectors cannot have nonzero sum.
+            if d == 0 or d is None:
+                return Integer(1)
+            else:
+                return Integer(0)
+
+        if d is None:
+            # No fixed sum, so sum could be anything up to k*m
+            dmax = k*m
+        else:
+            # Fixed sum
+            dmax = d
+
+        # Power series with enough precision that x^dmax is valid.
+        R = PowerSeriesRing(QQ, 'x', default_prec=dmax+1)
+        x = R.gen()
+
+        # Cardinality is computed using the Cycle Index Theorem.
+        # The figure-counting series, for maximum part size m, is
+        # (1-t**(m+1)) / (1-t) = 1+t+...+t**m.
+        # For function-counting series, we substitute x**cyclen for t.
+        #
+        # If there is no fixed sum requirement, we sum over the
+        # possible sums.
+        Z = G.cycle_index()
+        funcount = sum(
+            series_prod((1 - x**((m+1)*cyclen)) / (1 - x**cyclen)
+                        for cyclen in cyctype)
+            * coeff
+            for (cyctype, coeff) in Z)
```

Comment:
I don't mind much.  I read `cyclen` as a weird spelling of the plural of `cycle` (I'm Austrian, i.e. German speaking), and I was confused (even though I know my Pólya-Redfield quite well).  Since separating words with underscores is very common, I suggested it.  It is really as minor as it gets.

Reply:
I don't mind.  Use what fits you.

Reply:
Oh good. I was I afraid that the single letter is another house rule directed against code clarity. (You know, the compiler may be 2 nanoseconds faster that way, and you never know when someone is going to call the compiler a billion times in a tight loop.)

Reply:
I'm not joking (though I must admit I sometimes have difficulty keeping a straight face). I have seen so much terse SageMath code with single-letter variable names and nary a comment, that it did cross my mind that it is a house rule.

Reply:
I believe we can all agree that there are many (subjective) descriptions of what constitutes sufficient code clarity. Because of that, we can work together as a community without judgement to improve the overall utility of the software.